src/main/java/org/zalando/nakadi/repository/db/TimelineDbRepository.java

```diff
@@ -0,0 +1,129 @@
+package org.zalando.nakadi.repository.db;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+import org.zalando.nakadi.annotations.DB;
+import org.zalando.nakadi.domain.Timeline;
+
+@DB
+@Repository
+public class TimelineDbRepository extends AbstractDbRepository {
+
+    public static final String BASE_TIMELINE_QUERY = "SELECT " +
+            "   t_id, et_name, t_order, t.st_id, t_topic, " +
+            "   t_created_at, t_switched_at, t_cleanup_at, t_latest_position, " +
+            "   st_type, st_configuration " +
+            " FROM " +
+            "   zn_data.timeline t " +
+            "   JOIN zn_data.storage st USING (st_id) ";
+
+    @Autowired
+    public TimelineDbRepository(final JdbcTemplate jdbcTemplate, final ObjectMapper jsonMapper) {
+        super(jdbcTemplate, jsonMapper);
+    }
+
+    public List<Timeline> listTimelines(final String eventType) {
+        return jdbcTemplate.query(
+                BASE_TIMELINE_QUERY + " WHERE t.et_name=? order by t.t_order",
+                new Object[]{eventType},
+                timelineRowMapper);
```

Comment:
a very small impovement, but you can use the method notation when the mapper is on second place and params vararg is on 3rd place. That will allow to pass eventType without creation of array.